packages/ses/src/module-load.js

```diff
@@ -34,14 +34,52 @@ const resolveAll = (imports, resolveHook, fullReferrerSpecifier) => {
   return freeze(resolvedImports);
 };
 
+const loadRecord = async (
+  compartmentPrivateFields,
+  moduleAliases,
+  compartment,
+  moduleSpecifier,
+  staticModuleRecord,
+) => {
+  const { resolveHook, moduleRecords } = compartmentPrivateFields.get(
+    compartment,
+  );
+
+  // resolve all imports relative to this referrer module.
+  const resolvedImports = resolveAll(
+    staticModuleRecord.imports,
+    resolveHook,
+    moduleSpecifier,
+  );
+  const moduleRecord = freeze({
+    compartment,
+    staticModuleRecord,
+    moduleSpecifier,
+    resolvedImports,
+  });
+
+  // Memoize.
+  moduleRecords.set(moduleSpecifier, moduleRecord);
+
+  // Await all dependencies to load, recursively.
+  await Promise.all(
+    values(resolvedImports).map(fullSpecifier =>
+      // Behold: recursion.
```

Comment:
I wish "Behold: recursion." was the eslint directive ;)

Actually, I'll take this opportunity to ask: Do you think the Agoric style should turn this warning off by default?

Reply:
It’s a rare enough problem that it doesn’t bother me.

I don’t think I’ve ever used recursion accidentally, though.

The directive does force an opinionated order. In this particular case, these functions tend to tail-recur so they’re actually clearer to read in reverse declaration order.

I have no strong feelings.